File: mirrorbooth/ios/Runner/MirrorPlugin.swift
import Flutter
import Foundation

public class MirrorPlugin: NSObject, FlutterPlugin {
    let processor = MirrorVideoProcessor()
    private var channel: FlutterMethodChannel?

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "mirrorbooth/mirror",
            binaryMessenger: registrar.messenger()
        )
        let instance = MirrorPlugin()
        instance.channel = channel
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any] else {
            result(FlutterError(code: "INVALID_ARGS", message: nil, details: nil))
            return
        }
        switch call.method {
        case "setEnabled":
            processor.isEnabled = (args["enabled"] as? Bool) ?? false
            result(nil)
        case "setMirrorSide":
            let side = (args["side"] as? String) ?? "left"
            processor.mirrorLeft = side == "left"
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: mirrorbooth/android/app/src/main/kotlin/com/ol1n/mirrorbooth/MirrorPlugin.kt
package com.ol1n.mirrorbooth

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class MirrorPlugin : FlutterPlugin, MethodChannel.MethodCallHandler {
    private lateinit var channel: MethodChannel
    val processor = MirrorVideoProcessor()

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, "mirrorbooth/mirror")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "setEnabled" -> {
                val enabled = call.argument<Boolean>("enabled") ?: false
                processor.enabled.set(enabled)
                result.success(null)
            }
            "setMirrorSide" -> {
                val side = call.argument<String>("side") ?: "left"
                processor.mirrorLeft.set(side == "left")
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
